File: 1710-maximum-units-on-a-truck/1710-maximum-units-on-a-truck.js
/**
 * @param {number[][]} boxTypes
 * @param {number} truckSize
 * @return {number}
 */
var maximumUnits = function(boxTypes, truckSize) {
  boxTypes.sort(([b1, a], [b2, b]) => b - a)
  let rest = truckSize
  let ans = 0
  for (let [boxes, units] of boxTypes) {
    if (rest >= boxes) {
      ans += boxes * units
      rest -= boxes
    } else {
      ans += rest * units
      rest = 0
      return ans
    }
  }
  return ans
};

File: 1710-maximum-units-on-a-truck/1710-maximum-units-on-a-truck.go
func maximumUnits(boxTypes [][]int, truckSize int) int {
  sort.Slice(boxTypes, func(i, j int) bool {
    return boxTypes[i][1] > boxTypes[j][1]})
  total := 0
  for _, v := range boxTypes {
    if v[0] < truckSize {
      total += v[0] * v[1]
      truckSize -= v[0]
    } else {
      total += truckSize * v[1]
      break
    }
  }
  return total
}